src/java.base/share/classes/java/lang/reflect/Method.java

```diff
@@ -567,12 +567,42 @@ public Object invoke(Object obj, Object... args)
     }
 
     /**
-     * Returns {@code true} if this method is a bridge
-     * method; returns {@code false} otherwise.
+     * {@return {@code true} if this method is a bridge
+     * method; returns {@code false} otherwise}
      *
-     * @return true if and only if this method is a bridge
-     * method as defined by the Java Language Specification.
+     * @apiNote
+     * A bridge method is a {@linkplain isSynthetic synthetic} method created by a Java
+     * compiler alongside a method originating from the source
+     * code.
+     * Bridge methods are used by Java compilers in various
+     * circumstances to span across differences in Java programming
+     * language semantics and JVM semantics.
+     *
+     * One example use of bridge methods is as technique for a
+     * Java compiler to support <i>covariant overrides</i>, where a
+     * subclass overrides a method and gives the new method a more
+     * specific return type than the method in the superclass.
+     * While the Java language specification forbids a class
+     * declaring two methods with the same parameter types but a
+     * different return type, the virtual machine does not.
+     * A
+     * common case where covariant overrides are used is for a {@link
```

Comment:
I think the example should be clearer because here, you don't show the code of EnumSet.clone().
I wonder if it's not easier if all the code is visible
```
  interface I {
    Object m();
  }
  class A implements I {
    String m();
  }
```
so you can explain that the VM do the dispatch on I::m()Object so the compiler has to insert a method A::m()Object,
the bridge method with this pseudo-code
```
  class A implements I {
    /* bridge */ Object m() { return m(); } // calls m()String
    String m();
  }
```

Reply:
Hi Remi,

Thanks for the feedback. I did consider that kind of approach of an example from scratch. I judged referencing instances of bridge methods in the base module to be helpful to demonstrate it wasn't too esoteric of a feature in terms of it is something you, as a developer, may already have seen. Also, given the likely audience of the reading Class.isBridge, I didn't think a stand-alone example was warranted.

Reply:
The prose description here is still rather clumsy, though. If you're going to use EnumSet.clone() as an example, maybe run with that and list the methods directly instead of saying "EnumSet.clone() returns EnumSet rather than Object", be more explicit. Maybe something like the following:

The Object class has the following method:
```Object clone()```
whereas the EnumSet class has the following method:
```EnumSet<E> clone()```
From the JVM's perspective, the second method doesn't override the first, because they have different return types. The Java compiler inserts the following bridge method into the EnumSet class:
```Object clone()```
that simply calls the second method and returns its result.

Reply:
@jddarcy, the problem i see is that `EnumSet` is not used a lot and `clone` has another layer of complexity because of CloneNotSupportedException, etc.
So i'm not sure that using EnumSet.clone() is the right example here.

Perhaps String.compareTo()/Comparable.compareTo() is a better example ? But generics are in the middle too.